tools/walletextension/main/main.go

```diff
@@ -3,25 +3,47 @@ package main
 import (
 	"encoding/json"
 	"fmt"
+	"net"
+	"time"
 
 	"github.com/obscuronet/go-obscuro/tools/walletextension"
 )
 
 const (
+	tcp       = "tcp"
 	localhost = "127.0.0.1"
 )
 
 func main() {
 	config := parseCLIArgs()
+
+	// We wait thirty seconds for a connection to the node. If we cannot establish one, we exit the program.
+	fmt.Printf("Waiting up to thirty seconds for connection to host at %s...\n", config.NodeRPCHTTPAddress)
+	counter := 30
+	for {
+		conn, err := net.Dial(tcp, config.NodeRPCHTTPAddress)
+		if conn != nil {
+			conn.Close()
+		}
+		if err == nil {
+			break
+		}
+
+		counter--
+		if counter <= 0 {
+			fmt.Printf("Could not establish connection to host at %s. Exiting.", config.NodeRPCHTTPAddress)
```

Comment:
Need a newline character

Reply:
If you give it a spin, don't you automatically get a new line added on program return? I did on my machine.

Reply:
I didn't on mine;

```
Waiting up to thirty seconds for connection to host at 127.0.0.1:13000...
Could not establish connection to host at 127.0.0.1:13000. Exiting.Morays-MacBook-Pro:main moraygrieve$ ls -l
```

Reply:
Ah weird, will add, thanks.